gog-galaxy update uninstall

commitay opened the pull request:
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Removed `pkgutil` uninstall (no `pkg` record to uninstall) and add zap.

vitorgalvao commented on Casks/gog-galaxy.rb:
```diff
@@ -10,6 +10,7 @@
 
   pkg "galaxy_client_#{version}.pkg"
 
-  uninstall pkgutil: "com.gog.galaxy.galaxy_client_#{version}.pkg",
```
But this means the uninstall is now incomplete. There must be something else the `pkg` does.

commitay replied:
I've tried correcting `pkgutil` to `com.gog.galaxy` but it doesn't uninstall anything.
```
./list_ids_in_pkg ~/Library/Caches/Homebrew/Cask/gog-galaxy--1.1.30.5.pkg 
com.gog.galaxy
```
```
./list_payload_in_pkg ~/Library/Caches/Homebrew/Cask/gog-galaxy--1.1.30.5.pkg 
```
```
./list_recent_pkg_ids 
com.vmware.tools.macos.pkg.files
```
```
pkgutil --files com.gog.galaxy
No receipt for 'com.gog.galaxy' found at '/'.
```
```
pkgutil --pkgs
com.apple.pkg.BaseSystemResources
com.apple.pkg.ChineseWordlistUpdate.14U1315
com.apple.pkg.CLTools_Executables
com.apple.pkg.CLTools_SDK_OSX1012
com.apple.pkg.CoreADI
com.apple.pkg.CoreFP
com.apple.pkg.DevSDK
com.apple.pkg.DevSDK_OSX1012
com.apple.pkg.EmbeddedOSFirmware
com.apple.pkg.Essentials
com.apple.pkg.FirmwareUpdate
com.apple.pkg.GatekeeperConfigData.14U2309
com.apple.pkg.iTunesAccess
com.apple.pkg.iTunesX
com.apple.pkg.MobileDevice
com.apple.pkg.MRT.14U2301
com.apple.pkg.OSXSDK10.12
com.apple.pkg.RemoteDesktopClient
com.vmware.tools.macos.pkg.files
```

vitorgalvao replied:
Right, but that means we then need to find out what exactly this installs and remove it. A quick search revealed no documentation from their side.

victorpopkov replied:
I also have verified if there was something else before actually approving this PR and there is definitely nothing more than `com.gog.galaxy` which doesn't do anything when using in `uninstall` stanza as @commitay has already pointed out.

commitay commented:
Suspicious package says it contains no files but runs 13 scripts that run a `.tar.gz`, `.pkg`, and some fonts.

Besides the contents of the `.app` bundle this seems to be everything it installs.

```
/Library/LaunchAgents/com.gog.galaxy.commservice.plist
/Library/LaunchDaemons/com.gog.galaxy.ClientService.plist
/Library/PrivilegedHelperTools/com.gog.galaxy.ClientService
~/Library/Preferences/com.gog.galaxy.cef.renderer.plist
~/Library/Preferences/com.gog.galaxy.plist

/Users/Shared/GOG.com/Galaxy
/Users/Shared/GOG.com/Galaxy/Logs/GalaxyClient.log
/Users/Shared/GOG.com/Galaxy/redists/com.gog.galaxy.commservice.plist
/Users/Shared/GOG.com/Galaxy/redists/GalaxyCommunication
/Users/Shared/GOG.com/Galaxy/redists/GalaxyUpdater.app
/Users/Shared/GOG.com/Galaxy/redists/GalaxyUpdater.app/Contents/MacOS/GalaxyUpdater
/Users/Shared/GOG.com/Galaxy/redists/GalaxyUpdater.app/Contents/Resources/Galaxy.icns
/Users/Shared/GOG.com/Galaxy/redists/libGalaxyPeer.dylib
/Users/Shared/GOG.com/Galaxy/redists/libGalaxyPeer64.dylib
```

I missed the `plists` and `helpertool`, I've added them to the `delete:`

Changed `zap` to `rmdir:` `/Users/Shared/GOG.com` and added `delete:` `/Users/Shared/GOG.com/Galaxy`

victorpopkov commented on Casks/gog-galaxy.rb:
```diff
@@ -10,6 +10,14 @@
+  uninstall delete: [
+                      '/Applications/GalaxyClient.app',
+                      '/Library/LaunchAgents/com.gog.galaxy.commservice.plist',
+                      '/Library/LaunchDaemons/com.gog.galaxy.ClientService.plist',
```
You can use `launchctl` to remove `com.gog.galaxy.commservice.plist` and `com.gog.galaxy.ClientService.plist`.

In addition, I would also move other files away from `uninstall` back to `zap`. Especially the `/Users/Shared/GOG.com` directory to avoid users to reauthenticate if they decide to reinstall the cask:

```ruby
  uninstall delete:    '/Applications/GalaxyClient.app',
            launchctl: [
                         'com.gog.galaxy.ClientService',
                         'com.gog.galaxy.commservice',
                       ]

  zap delete: [
                '/Library/PrivilegedHelperTools/com.gog.galaxy.ClientService',
                '/Users/Shared/GOG.com',
                '~/Library/Preferences/com.gog.galaxy.cef.renderer.plist',
                '~/Library/Saved Application State/com.gog.galaxy.savedState',
              ]
```

commitay replied:
Thank you for checking @victorpopkov, I forgot about `launchctl`. I'll update the cask as you suggest.

vitorgalvao commented:
Thank you for all the research in making `uninstall` more complete.
